design/specs/specs-null-values.md

```diff
@@ -0,0 +1,124 @@
+
+## Prototype
+
+### Link
+
+[Prototype](https://mathesar-prototype.netlify.app/?activeTable=2)
+
+### Videos
+
+- Scenario 1: [Video](https://www.loom.com/share/527224982d534a2db082359360147ad4)
+- Scenario 2: [Video](https://www.loom.com/share/7de3c90af3d74ac195397c643f318867)
+- Scenario 3: [Video](https://www.loom.com/share/7ea4ca9fe3e14acdbffac5ea8e77e760)
+
+## User Experience
+
+### Considerations for handling NULL Values
+
+When presented in the form of tabular data, NULL values should be visually distinct so that the user understands that they are not the same as other string types. The style should attempt to communicate the absence of a value, similar to how placeholders are used in text input fields.
+As for the process of transforming values to and from NULL, it will be important to have a consistent way of doing this across different data types.
+
+## Scenarios
+
+### Scenario 1: User identifies a NULL Value
+
+#### Scenario 1a: The user identifies a NULL value in a table
+
+#### Steps for 1a
+
+1. The user opens a table that contains NULL values
+2. The user identifies a cell that contains a NULL value
+    - The cell contains the word NULL in all uppercase letters
+    - The cell has a lighter background-color
+    - The cell has a muted text color
+    - The cell has a different font style (italic)
+
+#### Scenario 1b: The user identifies a NULL value in a record preview
+
+#### Steps for 1b
+
+1. The user edits a value in a column that links to records from another table
+2. The user sees the list of records displayed in the record selection dropdown
+3. The user identifies a record that contains a NULL value
+    - The field contains the word NULL in all uppercase letters
+    - The text has a muted text color
+    - The text has a different font style (italic)
+
+### Scenario 2: User edits a NULL Value
+
+#### Scenario 2a: The data type is text
+
+#### Steps for 2a
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell and enters edit mode
+3. The user enters a value and leaves edit mode by hitting enter or clicking outside of the active cell
+4. The NULL value is replaced by the new value
+
+#### Scenario 2b: The data type is text, and the value is empty
+
+#### Steps for 2b
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell and enters edit mode
+3. The user doesn't enter a value and leaves edit mode by hitting enter or clicking outside of the active cell
+4. The NULL value is replaced by an empty string
+
+#### Scenario 2c: The data type is boolean with checkbox display enabled
+
+#### Steps for 2c
+
+1. The user edits a NULL value in a column that is set to boolean type with the checkbox display property enabled
+    - The field contains a checkbox in an indeterminate state
```

Comment:
It is not very clear to me what the indeterminate state represents looking at the UX.

* I think it would be better to show NULL next to the checkbox. Or,
* It might be better to show just NULL similar to other types, and only show the checkbox during focusing on the cell.

Reply:
I had the same thought initially. However I came to accept the idea of the indeterminate checkbox for the following reasons:

- Showing some kind of checkbox is nice because it allows the user to change the value from NULL to TRUE.
- Showing "NULL" next to an indeterminate checkbox would increase the column width. Perhaps that's okay though.

For those reasons, I chose to suggest a hover tooltip.

I'm still somewhat on the fence about this.

Reply:
I agree with @seancolsen here. I think showing `NULL` on hover will quickly teach users what an indeterminate checkbox is, even if the user doesn't get it at the first glance.